test/unit_conv2d.ml

```diff
@@ -0,0 +1,354 @@
+      [|1;2;1;1|] [|1;1|] VALID expected
+
+  (* conv2D2x2KernelDepth1 *)
+  let fun02 () =
+    let expected = [|5.;8.;14.;17.|] in
+    verify_value test_conv2d_back_kernel [|1;2;3;1|]
+      [|2;2;1;1|] [|1;1|] VALID expected
+
+  (* conv2D2x2Kernel *)
+  let fun03 () =
+    let expected = [|
+      17.; 22.; 27.; 22.; 29.; 36.; 27.; 36.; 45.; 32.; 43.; 54.; 37.; 50.; 63.;
+      42.; 57.; 72.; 62.; 85.; 108.; 67.; 92.; 117.; 72.; 99.; 126.; 77.; 106.;
+      135.; 82.; 113.; 144.; 87.; 120.; 153.|] in
+    verify_value test_conv2d_back_kernel [|1;2;3;3|]
+      [|2;2;3;3|] [|1;1|] VALID expected
+
+  (* conv2D1x2KernelStride3Width5 *)
+  let fun04 () =
+    let expected = [|9.;12.|] in
+    verify_value test_conv2d_back_kernel [|1;1;5;1|]
+      [|1;2;1;1|] [|3;3|] VALID expected
+
+  (* conv2D1x2KernelStride3Width6 *)
+  let fun05 () =
+    let expected = [|9.;12.|] in
+    verify_value test_conv2d_back_kernel [|1;1;6;1|]
+      [|1;2;1;1|] [|3;3|] VALID expected
+
+  (* conv2D1x2KernelStride3Width7 *)
+  let fun06 () =
+    let expected = [|9.;12.|] in
+    verify_value test_conv2d_back_kernel [|1;1;7;1|]
+      [|1;2;1;1|] [|3;3|] VALID expected
+
+  (* conv2D1x3Kernel *)
+  let fun07 () =
+    let expected = [|5.; 8.; 11.|] in
+    verify_value test_conv2d_back_kernel [|1;1;4;1|]
+      [|1;3;1;1|] [|1;1|] VALID expected
+
+  (* conv2D1x3Kernel *)
+  let fun08 () =
+    let expected = [|5.; 8.; 11.|] in
+    verify_value test_conv2d_back_kernel [|1;1;4;1|]
+      [|1;3;1;1|] [|1;1|] VALID expected
+
+
+  (* conv2D1x3KernelSame *)
+  let fun09 () =
+    let expected = [|20.; 30.; 20.|] in
+    verify_value test_conv2d_back_kernel [|1;1;4;1|]
+      [|1;3;1;1|] [|1;1|] SAME expected
+
+  (* conv2D1x3KernelSameOutbackprop2 *)
+  let fun10 () =
+    let expected = [|7.; 10.; 3.|] in
+    verify_value test_conv2d_back_kernel [|1;1;4;1|]
+      [|1;3;1;1|] [|2;2|] SAME expected
+
+  (* conv2D2x2KernelC1Same *)
+  let fun11 () =
+    let expected = [|91.; 58.; 32.; 17.|] in
+    verify_value test_conv2d_back_kernel [|1;2;3;1|]
+      [|2;2;1;1|] [|1;1|] SAME expected
+
+  (* conv2D2x2KernelStride2 *)
+  let fun12 () =
+    let expected = [|92.; 102.; 112.|] in
+    verify_value test_conv2d_back_kernel [|1;3;5;1|]
+      [|1;3;1;1|] [|2;2|] VALID expected
+
+  (* conv2D2x2KernelStride2Same *)
+  let fun13 () =
+    let expected = [|7.; 2.; 16.; 5.|] in
+    verify_value test_conv2d_back_kernel [|1;2;3;1|]
+      [|2;2;1;1|] [|2;2|] SAME expected
+
+end
+
+(* the tests *)
+
+let fun_conv00 () =
+  Alcotest.(check bool) "fun_conv00" true (To_test_conv2d.fun00 ())
+let fun_conv01 () =
+  Alcotest.(check bool) "fun_conv01" true (To_test_conv2d.fun01 ())
+let fun_conv02 () =
+  Alcotest.(check bool) "fun_conv02" true (To_test_conv2d.fun02 ())
+let fun_conv03 () =
+  Alcotest.(check bool) "fun_conv03" true (To_test_conv2d.fun03 ())
+let fun_conv04 () =
+  Alcotest.(check bool) "fun_conv04" true (To_test_conv2d.fun04 ())
+
```

Comment:
still, give one line space to separate functions. if you want to group them into different groups, you can add inline comments e.g. (* inverse functions ... *) and so on ...

Reply:
have you fixed all these line space issue?

Reply:
Just pushed the new commit.